src/AzureStack/Subscriptions/Subscriptions.Tests/src/Helpers/AzureStackTestBase.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using System;
+using System.Net;
+
+using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+
+namespace Subscriptions.Tests
+{
+    /// <summary>
+    /// Base class for all AzureStack tests.  This class should not be modified but extended through a child class.
+    /// </summary>
+    ///
+    public abstract class AzureStackTestBase<T> where T : class
+    {
+
+        /// <summary>
+        /// Reference to service client.
+        /// </summary>
+        protected T client;
+
+        /// <summary>
+        /// Validate the client.
+        /// </summary>
+        /// <param name="client">The instantiated client we want to validate.</param>
+        protected abstract void ValidateClient(T client);
+
+        /// <summary>
+        /// The default location for all admin actions.  Override in derived class as needed.
+        /// </summary>
+        protected string Location = "local";
+
+        /// <summary>
+        /// Run a test that accepts no arguments.  An exception can be 
+        /// </summary>
+        /// <param name="test">The test we wish to run.</param>
+        /// <param name="before">Function to execute before your test has completed.</param>
+        /// <param name="after">Function to execute after your test has completed.</param>
+        /// <param name="status">Expected returned HttpStatusCode.</param>
+        /// <exception cref="System.Exception">Thrown when an unexpected exception occurs.</exception>  
+        ///
+        protected void RunTest(Action<T> test,
+            Action before = null,
+            Action after = null,
+            HttpStatusCode status = HttpStatusCode.OK,
+            [System.Runtime.CompilerServices.CallerMemberName]
+            string methodName= "testframework_failed") {
+
+            Exception caught = null;
+
+            try
+            {
+                var handler = new RecordedDelegatingHandler { StatusCodeToReturn = status };
+                handler.IsPassThrough = true;
+
+                using (MockContext context = MockContext.Start(typeof(T).Name, methodName))
+                {
+                    var client = context.GetServiceClient<T>(handlers: handler);
+                    ValidateClient(client);
+
+                    before?.Invoke();
+                    test(client);
+                    after?.Invoke();
+                }
+            }
+            catch (Exception ex)
+            {
+                caught = ex;
+            }
+            finally
+            {
+                if (caught != null)
```

Comment:
Why aren't we just rethrowing? We don't use caught in try.

Reply:
I think a bad copy/paste from some other file.. we should rethrow here.. fixing